internal/controller/openstackclusterstackrelease_controller.go

```diff
@@ -142,6 +127,21 @@ func (r *OpenStackClusterStackReleaseReconciler) Reconcile(ctx context.Context,
 		// acquire lock so that only one reconcile loop can download the release
 		r.openStackClusterStackRelDownloadDirectoryMutex.Lock()
 
+		gc, err := r.GitHubClientFactory.NewClient(ctx)
+		if err != nil {
+			conditions.MarkFalse(openstackclusterstackrelease,
+				apiv1alpha1.GitAPIAvailableCondition,
+				apiv1alpha1.GitTokenOrEnvVariableNotSetReason,
+				clusterv1beta1.ConditionSeverityError,
+				err.Error(),
+			)
+			record.Warnf(openstackclusterstackrelease, "GitTokenOrEnvVariableNotSet", err.Error())
+			logger.Error(err, "failed to create Github client")
+			return ctrl.Result{}, nil
+		}
+
+		conditions.MarkTrue(openstackclusterstackrelease, apiv1alpha1.GitAPIAvailableCondition)
```

Comment:
minor: 
- maybe could be a better idea to create a GitHub client before `r.openStackClusterStackRelDownloadDirectoryMutex.Lock()`. 
- the reason is that the GitHub client creation could fail, hence the lock is not needed then

Reply:
that is indeed true and a big problem! Thanks for bringing this up also related to cso! We should update this in both repos!

Reply:
See abeed44724d3152234b9ac948af8d8a53e757a6b